README.md

```
@@ -31,21 +32,110 @@ This repo contains the required AWS resources for doing a transfer using PSTN on
+
+**Validation Rules for `apiDomain` and `authDomain`:**
+
+- **OAuth with `apiDomain`**: If you're using OAuth authentication (`oauthClientId`/`oauthClientSecret` or `oauthSecretArn`) and provide `apiDomain`, then `authDomain` **must** also be provided.
+- **API key with `apiDomain`**: If you're using API key authentication (`apiKey`) and provide `apiDomain`, `authDomain` is **not required** (API key doesn't use OAuth endpoints).
+- **OAuth without `apiDomain`**: If you're using OAuth authentication but don't provide `apiDomain`, `authDomain` is **not required** (the system will automatically build the auth domain from the `region` parameter).
+
 - **virtualAgentName**: The resourcename of the virtual agent the call is transferred to. Format: `customers/{customer}/profiles/{profile}/virtualAgents/{virtualAgentID}`. Required.
 
+#### Examples
+
+**Example 1: Using standard region with OAuth**
```

markdownlint-cli2 (0.18.1)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)